store/tikv/client_batch.go

```diff
@@ -193,26 +199,59 @@ func (a *batchConn) isIdle() bool {
 	return atomic.LoadUint32(&a.idle) != 0
 }
 
+func (a *batchConn) callbackResponse(resp *tikvpb.BatchCommandsResponse) {
+	if resp == nil {
+		a.failPendingRequests(context.Canceled)
```

Comment:
Return cancel in here maybe cause 

https://github.com/pingcap/tidb/blob/a53cf420759d12c626b8d14e0b1c941429b8cbef/store/tikv/region_request.go#L555

return error to user directly without retry, keep origin error from `RecvLoop` goroutine should be easy way